scripts/test-remote-dev-context7-device-login.py

```
@@ -0,0 +1,550 @@
+            else:
+                raise AssertionError("Context7 login accepted a failed transient-state cleanup")
+        finally:
+            module.run_login_process = original_login_process
+            module.RUN_ROOT = original_run_root
+            module.NPM = original_npm
+            module.MISE_CONFIG = original_mise_config
+            module.SETPRIV = original_setpriv
+            module.shutil.rmtree = original_rmtree
+
+
+def assert_credentials_contract(module) -> None:
```

Ruff (0.16.1)

[warning] Too many statements (59 > 50)

(PLR0915)